### PR TITLE
[IN-107] Feat/in107 create selection page

### PR DESCRIPTION
**Pull request by ghost**
#### User Story

- https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1?selectedIssue=IN-107
- Rene and I both worked on this together.

#### Changes made

- On selection page, can click preview prefills, which takes you to a new Template page
- On this page, users can see the prefills (technical requirements specifically) templates
- Adding templateObjects, these will be passed into Create to pre-populate the jobData object with the templates (also includes a blankStudent which is what is used when you create a new posting)
- onClick feature added to each template div, can support extendable functionality (behaviour on click)
- Back button takes you back to selection page (preview prefills/make new posting)

#### Does your new code introduce new warnings on dev console?
No

#### Test Steps
Go to localhost:3000/templates, click on each, look inside console.

### CODE REVIEW

File: client/src/components/pages/Templates.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { TemplatePosting } from "../molecules/index";
+import {
+  backEndStudent,
+  frontEndStudent,
+  dataScienceStudent,
+  fullStackStudent,
+} from "../../models/templateJobDataObjects";
+import "./styles/Template.css";
+
+function dummyfunction() {
```
Comment (JudeShamsi) [inline client/src/components/pages/Templates.js]: pls remove

File: client/src/components/pages/Templates.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        container
+        style={{ paddingTop: "1em" }}
+        direction="row"
+        alignItems="flex-start"
+        justifyContent="flex-end"
+      >
+        <Grid item xs={10}>
+          <Grid container direction="row">
+            <div className="templates_container">
+              <div className="template" onClick={() => dummyfunction()}>
```
Comment (JudeShamsi) [inline client/src/components/pages/Templates.js]: pls remove for all these divs

**Review by JudeShamsi — CHANGES_REQUESTED** [review]
we shouldn't be adding dummy functions or un-used functionality in develop

**Review by JudeShamsi — APPROVED** [review]

File: client/src/components/molecules/TemplatePosting.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          </ul>
+        </ul>
+      </div>
+    );
+  };
+
+  return (
+    <>
+      <JobPosting />
+    </>
```
Comment (tommytwm) [inline client/src/components/molecules/TemplatePosting.js]: the empty div is unecessary if you're only returning `<JobPosting />`

File: client/src/components/molecules/ViewPosting.js [hunk 4]
```diff
@@ -161,9 +161,9 @@ const ViewPosting = (props) => {
     <>
       <JobPosting />
       <div className="view_posting_submit">
-        <ButtonFilled startIcon={<PictureAsPdfIcon />} onClick={handleClick}>
+        {props.hide? null : <ButtonFilled startIcon={<PictureAsPdfIcon />} onClick={handleClick}>
```
Comment (tommytwm) [inline client/src/components/molecules/ViewPosting.js:164]: run the Prettier formatter

**Review by tommytwm — CHANGES_REQUESTED** [review]
Double check your class names, there's a conflict that's causing the navbar padding to shrink and now there's awkward spacing: 
image

**Comment by tommytwm** [comment]
Remove this grey line (it's supposed to be the scroll bar)
image

**Comment by tommytwm** [comment]
Remove the max height on this container so that users are able to scroll vertically without the view being fixed height
image

**Comment by ghost** [comment]
> Double check your class names, there's a conflict that's causing the navbar padding to shrink and now there's awkward spacing:
> image

I think this is an issue on develop. I'm encountering the same thing on my branch (merged develop into it), and it doesn't contain the templates. We'll see what we can do about fixing it in this PR though.

**Comment by tommytwm** [comment]
> > Double check your class names, there's a conflict that's causing the navbar padding to shrink and now there's awkward spacing:
> > image
> 
> I think this is an issue on develop. I'm encountering the same thing on my branch (merged develop into it), and it doesn't contain the templates. We'll see what we can do about fixing it in this PR though.

Seems like this is a one-off thing. I've tried replicating this again by restarting the app locally on both develop and this branch, and I wasn't able to replicate it. We can discard this if we're not seeing this again

**Comment by tommytwm** [comment]
I'm not getting the slightly dark grey document on hover. Is this implemented?

**Comment by tommytwm** [comment]
I'm also not able to click on the templates to go to the create steps. Is this replicable?

File: client/src/components/pages/Templates.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+              </div>
+              <div className="template" >
+                <TemplatePosting data={frontEndStudent} hide={true}/>
+              </div>
+              <div className="template" >
+                <TemplatePosting data={dataScienceStudent} hide={true}/>
+              </div>
+              <div className="template">
+                <TemplatePosting data={fullStackStudent} hide={true}/>
+              </div>
```
Comment (tommytwm) [inline client/src/components/pages/Templates.js:53]: I'm not seeing an onClick callback that would bring the user to the Create step and pass in the relevant information. Is there a call to the jobs reducer that passes in the relevant information as well?

**Review by tommytwm — CHANGES_REQUESTED** [review]

File: client/src/components/pages/Templates.js (hunk 5 above)
Comment (ReneReid) [reply client/src/components/pages/Templates.js:53]: @tommytwm please see Jira tickets - @JudeShamsi will be implementing this.
https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1/backlog?selectedIssue=IN-100

**Comment by ghost** [comment]
> I'm not getting the slightly dark grey document on hover. Is this implemented?

So unfortunately, I don't think this can be added without doing a lot more work. Because of the scrollbar, when you set the background to dark grey on a hover, it only sets the part of the div that is in the scrollbar's view to grey, the rest is white even when you scroll down.

**Pull request closed without merging**